spec/models/service_provider_spec.rb

```diff
@@ -15,6 +15,29 @@
     end
   end
 
+  describe 'scopes' do
+    before do
+      clear_agreements_data
+      Agency.destroy_all
+      ServiceProvider.destroy_all
+    end
+
+    let!(:external_sps) { create_list(:service_provider, 2, iaa: 'LG1234') }
```

Comment:
since you made the `external` trait for the factory, can you just use that rather than setting the iaa here?

`let!(:external_sps) { create_list(:service_provider, 2, :external) }`

Reply:
Oh yeah great catch!
```suggestion
    let!(:external_sps) { create_list(:service_provider, 2, :external) }
```